docs/platform-engineers/addon/intro.md

```diff
@@ -100,175 +107,493 @@ metadata:
+After rendered, the resulting application is:
 
-Addon is actually an Application, but it provides more convenience for you to compose a complex application. You can use CUE or YAML to render resources into an application. This is what `resources` directory does for you.
+```yaml
+kind: Application
+metadata:
+  name: example
+  namespace: vela-system
+spec:
+  components:
+    - name: namespace
+      type: k8s-objects
+      properties:
+        objects:
+          - apiVersion: v1
+            kind: Namespace
+            metadata:
+              name: my-namespace
+```
 
-Files defined in `resources` folder eventually rendered as components in the application described in `template.yaml`.
+> Please notice: The name of Application in template file will be replaced by the addon name in `metadata.yaml`. The addon application will always have a unified name in the format of `addon-<addon_name>`.
 
-#### CUE format resource
+Although you can define the entire content of the application in the template file, but this may cause the template file to become very large, so you can choose to split a part of some content into the `resources` directory, latter section will introduce the all details.
 
-If you need to add a component to your application that needs to be rendered dynamically based on parameters when enabled, you can write a CUE format file as shown below:
+##### Auxiliary resources
+
+You also can define some auxiliary resources that only to applied to control-plane in `outputs` CUE block of `template.cue` file. These resources will applied to control-plane directly. For example:
 
 ```cue
+package main
+
 output: {
-	type: "webservice"
-	properties: {
-		image: "oamdev/vela-apiserver:v1.4.0"
+	apiVersion: "core.oam.dev/v1beta1"
+	kind:       "Application"
+	metadata: {
+		name:      "example"
+		namespace: "vela-system"
 	}
-	traits:[{
-		type: "service-account"
-		properties: name: "serviceAccountName"
-	}]
+	spec: {
+		
+	}
+	... 
 }
+
+outputs: resourceTree: {
+	apiVersion: "v1"
+	kind:       "ConfigMap"
+	metadata: {
+		name:      "resource-tree"
+		namespace: "vela-system"
+		labels: {
+			"rules.oam.dev/resources":       "true"
+			"rules.oam.dev/resource-format": "json"
+		}
+	}
+	data: rules: json.Marshal(_rules)
+}
+
+_rules: {...}
 ```
 
-You can refer to the [CUE basic](../cue/basic) to learn language details.
+In this example, we define a configmap `resourceTree` as an auxiliary resource, this configmap is actually a [resource topology rule](../../reference/topology-rule) that only need to be applied to control-plane.
+
+##### Determine which clusters to be installed by parameters in `template.cue`
+
+If you want the resources in the addon to be installed not only in the control-plane, but also in various other managed-clusters, and installed clusters need to be specified by the user by setting the `clusters` parameter when enabling the addon. You can do this by adding a topology policy in `template.cue` as shown below:
```

Comment:
If you want the resources in the addon to be installed not only in the control-plane, but also in managed clusters,  you can use the `topology` policy in your application as shown below: